src/data/s3-data-source.ts

```diff
@@ -64,12 +65,57 @@ export class S3DataSource implements ContiguousDataSource {
     });
 
     try {
+      // TODO: Use S3 client instead of accessing  aws client directly when aws-lite s3 supports Metadata on head-requests
+      // const head = this.s3Client.HeadObject({
+      //   Bucket: this.s3Bucket,
+      //   Key: `${this.s3Prefix}/${id}`,
+      // });
+
+      log.debug('Fetching S3 metadata', {
+        id,
+        bucket: this.s3Bucket,
+        prefix: this.s3Prefix,
+      });
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Actually, one more question... :sweat_smile: 

Can we pass in `awsClient` from `systemt.ts` in via the constructor and get rid of this?

Reply:
49319eb